## Description: Add package-extract utility.
## Origin/Author: Lenik (xjl@99jsj.com)
Index: hex-a-hop-0.0.20070315/Makefile
===================================================================
--- hex-a-hop-0.0.20070315.orig/Makefile	2010-12-30 10:35:52.309700002 +0800
+++ hex-a-hop-0.0.20070315/Makefile	2010-12-30 10:35:52.569700002 +0800
@@ -13,6 +13,8 @@

 OBJS=$(CXXSOURCES:.cpp=.o)

+all : $(NAME)
+
 %.o	: %.cpp
	$(GCC) $(CXXFLAGS) -D_VERSION=\"$(VERSION)\" -DDATA_DIR=\"$(DATA_DIR)\" `sdl-config --cflags`  -c -o $@ $<

Index: hex-a-hop-0.0.20070315/debian/install
===================================================================
--- hex-a-hop-0.0.20070315.orig/debian/install	2010-12-30 10:35:49.913700002 +0800
+++ hex-a-hop-0.0.20070315/debian/install	2010-12-30 10:36:01.033700001 +0800
@@ -10,3 +10,4 @@
 graphics/title.dat usr/share/games/hex-a-hop/graphics/
 debian/*.xpm usr/share/pixmaps/
 debian/*.desktop usr/share/applications/
+package-extract usr/bin
Index: hex-a-hop-0.0.20070315/package-extract
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ hex-a-hop-0.0.20070315/package-extract	2010-12-30 10:35:52.569700002 +0800
@@ -0,0 +1,34 @@
+#!/usr/bin/python
+
+import struct
+import sys
+import os
+
+from os import path
+
+fname = sys.argv[1]
+with open(fname, 'rb') as f:
+    while True:
+        _size = f.read(4)
+        if _size.__len__() == 0:
+            break
+        size = struct.unpack('<i', _size)
+        size = size[0]
+
+        block = f.read(size)
+
+        p = block.find('\0')
+        if p == -1:
+            continue
+
+        entry = block[:p].replace('\\', '/')
+        block = block[p + 1:]
+
+        print entry
+        dirname = path.dirname(entry)
+        if not path.exists(dirname):
+            os.makedirs(dirname)
+
+        with open(entry, 'wb') as wf:
+            wf.write(block)
+
